### ISSUE TITLE
wrong Lint/UnusedBlockArgument when use eval

### ISSUE THREAD

**Issue by daixiang0** [bbatsov/rubocop#4677]
```
  if ovals
    var_vals.zip(ovals).map do |var_val, oval| <===
      var, val = var_val
      eval("#{var} = oval")  <===
    end 
  end 
```
In fact, i use oval val but rubocop say it does not be used.

**Comment by pocke** [comment]
`Lint/UselessAssignment` and `Lint/UnusedMethodArgument` have same false positives.
For example:


```ruby
def lvar
  x = 1
  eval str
end

def arg(x)
  eval str
end

block do |x|
  eval str
end
```


```bash
$ rubocop -D --only Lint
Inspecting 1 file
W

Offenses:

test.rb:2:3: W: Lint/UselessAssignment: Useless assignment to variable - x.
  x = 1
  ^
test.rb:6:9: W: Lint/UnusedMethodArgument: Unused method argument - x. If it's necessary, use _ or _x as an argument name to indicate that it won't be used. You can also write as arg(*) if you want the method to accept any arguments but don't care about them.
def arg(x)
        ^
test.rb:10:11: W: Lint/UnusedBlockArgument: Unused block argument - x. You can omit the argument if you don't care about it.
block do |x|
          ^

1 file inspected, 3 offenses detected
```


------


I cannot decide to make this cop aware of `eval`. We avoid the false positive by changing this cop, but false negatives occur by this change.


```ruby
block do |v|
  eval str # If str does not include `v`, rubocop should add an offense.
end
```

-------


Note: The cops do not add offense if the methods return `binding`.

```ruby
def lvar
  x = 1
  eval str
  binding
end

def arg(x)
  eval str
  binding
end

block do |x|
  eval str
  binding
end
```

```ruby
$ rubocop -D --only Lint
Inspecting 1 file
.

1 file inspected, no offenses detected
```
